test/fs.js

```diff
@@ -0,0 +1,186 @@
+'use strict';
+
+const metatests = require('metatests');
+const metasync = require('metasync');
+const fs = require('fs');
+const path = require('path');
+const stream = require('stream');
+const { copyDir } = require('..');
+
+const testDataFile =
+  'JySAF1figbONWxAeeoGhDXeI7df25GKXKPcyCl63ZjIv5HZGQH8yfFeHGGaSJ9zDMP0CwtgX2madIQkFmuq38SNhcKcGQ';
+const srcDir = 'sourceDirectory';
+const targetDir = 'targetDirectory';
+const absoluteLink = __dirname.replace('test', 'lib');
+const files = [
+  '1.js',
+  '2.txt',
+  '3.html',
+  '4.css',
+  '5.doc',
+  '6.tmp',
+  '7.asm',
+  '8.zip',
+  'link',
+  'dir1',
+  'dir2',
+  'dir3',
+];
+
+const createTestDir = (current, cb, curDepth = 1) => {
+  if (curDepth > 4) {
+    cb();
+    return;
+  }
+  fs.mkdir(current, err => {
+    if (err) {
+      cb(err);
+      return;
+    }
+    metasync.each(
+      files,
+      (file, cb) => {
+        const srcFile = path.join(current, file);
+        if (file.includes('.')) {
+          fs.appendFile(srcFile, testDataFile, cb);
+        } else if (file === 'link') {
+          fs.symlink(absoluteLink, path.join(current, file), cb);
+        } else {
+          createTestDir(srcFile, cb, curDepth + 1);
+        }
+      },
+      cb
+    );
+  });
+};
+
+const rmdirp = (dir, cb) => {
+  fs.lstat(dir, (err, stats) => {
+    if (err) {
+      if (err.code === 'ENOENT') cb(null);
+      else cb(err);
+      return;
+    }
+    if (stats.isDirectory()) {
+      fs.readdir(dir, (err, files) => {
+        if (err) {
+          cb(err);
+          return;
+        }
+        files = files.map(file => path.join(dir, file));
+        metasync.each(files, rmdirp, err => {
+          if (err) cb(err);
+          else fs.rmdir(dir, cb);
+        });
+      });
+    } else {
+      fs.unlink(dir, cb);
+    }
+  });
+};
+
+class ToUpperCase extends stream.Transform {
+  constructor(options = {}) {
+    options = Object.assign(options, {
+      decodeStrings: false,
+    });
+    super(options);
+  }
+  _transform(chunk, encoding, callback) {
+    if (encoding !== 'utf8') {
+      this.emit('error', new Error('only utf8'));
+      return callback();
+    }
+    this.push(chunk.toUpperCase());
+    return callback();
+  }
+}
+
+metatests.test('fs / copyDir', test => {
+  createTestDir(srcDir + '1', err => {
+    if (err) test.error(err);
```

Comment:
```suggestion
    if (err) {
      test.error(err);
      return;
    }
```
If test cannot be continued without previous operation being successful, there should be `return`.

Reply:
Also, there is now `test.cb` and `test.cbFail` in [metatests 0.5.0](https://github.com/metarhia/metatests) which can be used to reduce manual error handling.